.github/workflows/ci.yml

```diff
@@ -333,33 +226,42 @@ jobs:
           pwd
           echo "$(pwd)\winfsp-test" >> "$GITHUB_PATH"
 
-      - uses: ./.github/actions/setup-rust
-        if: steps.python-changes.outputs.run == 'true'
-        with:
-          version: ${{ env.rust-version }}
-          cache-key: integration
-          os: ${{ matrix.os }}
-        timeout-minutes: 10
-
       - uses: ./.github/actions/setup-python-poetry
         if: steps.python-changes.outputs.run == 'true'
         id: setup-python
         with:
           python-version: ${{ env.python-version }}
           poetry-version: ${{ env.poetry-version }}
 
-      - name: Install python deps
-        if: steps.python-changes.outputs.run == 'true'
-        run: poetry install -E core -E backend
+      # libparsec is slow to compile, so we save it in cache and skip the
+      # compilation entirely if the Rust code hasn't changed !
+
+      - name: Use cached libparsec if Rust hasn't been modified
+        id: cache-libparsec
+        if: >-
+          steps.python-changes.outputs.run == 'true'
+          && steps.rust-changes.outputs.run != 'true'
+        uses: actions/cache@v3
+        with:
+          # Key cache contains a hash of all the files that are used to produce _parsec.so
+          # Hence if we have a cache hit we know that there is no need for a rebuild !
+          key: ${{ runner.os }}-${{ hashFiles('src/**', 'oxidation/libparsec/**', 'rust-toolchain.toml', 'Cargo.toml', 'Cargo.lock') }}-libparsec-python
+          path: |
+            parsec/_parsec.*.pyd
+            parsec/_parsec.*.so
+
+      - name: Setup Rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@685abf830623a8456904376fce2de24302d98a7f # pin v1.3.4
+        if: steps.cache-libparsec.outputs.cache-hit != 'true'
 
-      - name: Debug python & pip version
+      - name: Install python deps
         if: steps.python-changes.outputs.run == 'true'
-        shell: bash
-        continue-on-error: true
         run: |
-          set -x
-          poetry run python --version
-          poetry run pip --version
+          if ${{ steps.cache-libparsec.outputs.cache-hit == 'true' }}; then export POETRY_LIBPARSEC_BUILD_STRATEGY=no_build; fi
+          poetry install -E core -E backend
+        shell: bash
+        env:
+          CARGO_PROFILE: ${{ env.CARGO_PROFILE }}
```

Comment:
no need to re-set it

Reply:
Fixed in [3440cc3](https://github.com/Scille/parsec-cloud/pull/3761/commits/3440cc3c7cd9bdc4550fb8f02d0a1daa842bc83b)

Reply:
done in 3440cc3

Reply:
See my comment https://github.com/Scille/parsec-cloud/pull/3761#discussion_r1050570988

I've even removed `CARGO_PROFILE` env to be extra explicit (which is good given now we have two different ci profiles !)